### DISCUSSION TITLE
Whether the default `content-type` should be implied?

### DISCUSSION THREAD

**Discussion by AlexV525** [cfug/diox#52] [Polls]
This should be addressed before the `diox` `5.0.0` stable release.

Previously in `dio`, the `content-type` will be implied to
* `application/json; charset=utf-8` by default.
This makes many applications behave normally without extra configurations, even without tweaking the `BaseOptions`.

However, this is not following any standard or specification, leading users that follow the specification confused and maintainers hard to fix related issues.

We should carefully discuss these:
1. Whether the `diox` should continue to imply the `content-type` in the future.
2. Whether the `BaseOptions.setRequestContentTypeWhenNoPayload` should be removed.

cc @cfug/devs

#### In dio

- https://github.com/flutterchina/dio/blob/9040bde4a07ffe4e201e4ec014242a16eb69d926/dio/lib/src/options.dart#L289-L297
- https://github.com/flutterchina/dio/blob/9040bde4a07ffe4e201e4ec014242a16eb69d926/dio/lib/src/options.dart#L614-L616

#### In diox

- https://github.com/cfug/diox/blob/580b08d3e4bb2cd71f7ca8a1cd93ae3705c8d7e4/dio/lib/src/options.dart#L306-L319
- https://github.com/cfug/diox/blob/580b08d3e4bb2cd71f7ca8a1cd93ae3705c8d7e4/dio/lib/src/options.dart#L663-L665
- https://github.com/cfug/diox/blob/2b7aa3bbe57df10307c30f3172270daa7ae7ff5c/dio/lib/src/transformer.dart#L79-L83

**Comment by kuhnroyal** [comment]
I am not sure, the Transformer encodes the JSON by default if the content type is set, right?

**Reply by AlexV525** [reply]
> the Transformer encodes the JSON by default if the content type is set, right?

Partially correct, see this:
https://github.com/cfug/diox/blob/dd6e39bdf03316216c0364ae211fc1f9847a7e1f/dio/lib/src/transformer.dart#L77-L78

**Comment by AlexV525** [comment]
I think we have the conclusion at present, we'll make no content type implied by default.
